Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -692,6 +692,94 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
         XCTAssertEqual(ObjectiveCSupport.convert(object: Credentials.anonymous),
                        RLMCredentials.anonymous())
     }
+    // MARK: - Bundled Sync Realm
+
+    func testSeedFilePathOpen() {
+        do {
+            // user1 creates and writeCopies a realm to be opened by another user
+            let user1 = try logInUser(for: basicCredentials())
+            var config = user1.configuration(testName: #function)
+
+            // This test uses the following instead of `populateRealm`
+            // for consistent objectTypes in CI testing.
+            config.objectTypes = [SwiftHugeSyncObject.self]
+            let realm = try Realm(configuration: config)
+            try! realm.write {
+                for _ in 0..<SwiftSyncTestCase.bigObjectCount {
+                    realm.add(SwiftHugeSyncObject.create())
+                }
+            }
+            waitForUploads(for: realm)
+
+            try realm.writeCopy(toFile: RLMTestRealmURL(), enableSync: true)
```

Comment:
You would receive an exception:
```
Incompatible histories. Expected a Realm with no or in-realm history, but found history type 3
```

Reply:
Can we test this and try give a better exception back to the user?

Reply:
We can't check history before the realm open. Offline decided to document what incorrect usage could receive "incompatible history" error.

Reply:
We can catch the IncompatibleHistory exception and throw a new one with a better error message. This should just require adding a clause to `RLMRealmTranslateException()`.

Reply:
What kind of clause would be added? Should I be looking for what `kind` the exception is?
I can't switch case on a new kind, since the exception is classified as `AccessError`, which already exists. What property in the error could I branch off of? `m_underlying` seems like the only unique thing? But that's just a string.

<img width="1653" alt="Screen Shot 2022-01-10 at 3 06 20 PM" src="https://user-images.githubusercontent.com/25092858/148832307-a4e49134-1af6-4211-9933-f23f1562c81d.png">

Reply:
It throws an `IncompatibleHistories` exception. If you catch that it'll take precedence over the catch clause for the base class.

Reply:
Or nevermind, it looks like it's being translated incorrectly in the object store code before that. We do check the string message in a few spots where there's no better way, but really the OS code should be adjusted to throw something better.

Reply:
Do you know a good spot where we check the string message? So that I can see how do it in other places.

Reply:
> it looks like it's being translated incorrectly in the object store code before that.

Where does this happen? So I can create a core ticket.

Reply:
I wasn’t sure the best way to check the underlying, so I committed a possible example to see what you think.

As an aside, the test I added started failing because of a session sticking around whenever I added string check. I couldn't tell why.
<img width="647" alt="Screen Shot 2022-01-10 at 4 13 40 PM" src="https://user-images.githubusercontent.com/25092858/148841214-9a3bbf29-163e-459c-a855-0e2fe02b3e46.png">